homeassistant/components/light/tradfri.py

```diff
@@ -162,42 +228,90 @@ def rgb_color(self):
         """RGB color of the light."""
         return self._rgb_color
 
-    def turn_off(self, **kwargs):
+    @asyncio.coroutine
+    def async_turn_off(self, **kwargs):
         """Instruct the light to turn off."""
-        self._api(self._light_control.set_state(False))
+        self.hass.async_add_job(self._api(
+            self._light_control.set_state(False)))
 
-    def turn_on(self, **kwargs):
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
         """
         Instruct the light to turn on.
 
         After adding "self._light_data.hexcolor is not None"
         for ATTR_RGB_COLOR, this also supports Philips Hue bulbs.
         """
-        if ATTR_BRIGHTNESS in kwargs:
-            self._api(self._light_control.set_dimmer(kwargs[ATTR_BRIGHTNESS]))
-        else:
-            self._api(self._light_control.set_state(True))
-
         if ATTR_RGB_COLOR in kwargs and self._light_data.hex_color is not None:
-            self._api(self._light.light_control.set_rgb_color(
-                *kwargs[ATTR_RGB_COLOR]))
+            self.hass.async_add_job(self._api(
+                self._light.light_control.set_rgb_color(
+                    *kwargs[ATTR_RGB_COLOR])))
 
         elif ATTR_COLOR_TEMP in kwargs and \
-                self._light_data.hex_color is not None and self._ok_temps:
+                self._light_data.hex_color is not None and \
+                self._temp_supported:
             kelvin = color_util.color_temperature_mired_to_kelvin(
                 kwargs[ATTR_COLOR_TEMP])
-            self._api(self._light_control.set_kelvin_color(kelvin))
+            self.hass.async_add_job(self._api(
+                self._light_control.set_kelvin_color(kelvin)))
+
+        keys = {}
+        if ATTR_TRANSITION in kwargs:
+            keys['transition_time'] = int(kwargs[ATTR_TRANSITION])
```

Comment:
keys['transition_time'] = int(kwargs[ATTR_TRANSITION] * 10)

Reply:
Also it would be nice to set a default 'transition_time' (possibly 1 sec?) This would enable transition effect when brightness is adjusted directly via the slider.

Reply:
@javefang Personally I don't like the idea of default transition time. But I noticed, after decompiling, that the default transition time in the ikea tradfri android app is 0.5 seconds.

Found in: com.ikea.tradfri.lighting.ipso.Light
private int transition_time = 5;

Reply:
I came from the Hue platform, which always have a smooth transition on brightness change. In the end it is probably down to personal preference, so it feels like it would be better if we either keeping it consistent with the app (0.5 sec), or make it configurable?